activerecord/lib/active_record/migration.rb

```diff
@@ -621,6 +621,17 @@ def build_watcher(&block)
         def connection
           ActiveRecord::Base.connection
         end
+
+        def check_pending_migrations_unsafely
+          prev_db_config = Base.connection_db_config
+          all_configs = ActiveRecord::Base.configurations.configs_for # this needs an env_name parameter but i can't figure it out
```

Comment:
You can grab the ENV from `ActiveRecord::ConnectionHandling::DEFAULT_ENV.call`.

Reply:
Oh, after looking at its definition I see why it's used in a few places!